README.md

```diff
@@ -0,0 +1,320 @@
+
+## Code Walkthrough 
+
+### Server ###
+
+The app.js server file consists of a function, which when called instantiates an express application, then mounts helper functions and Router instances.
+Helper functions consist of bodyparsing and HTTP request logger middleware.
+It contains final error-handling middleware function to send error responses. 
+Finally, it sets the server up at specified port 4001 listening for requests.
+
+The app function is called in buildServer.js, passing in an id generator - shortid helper node module. 
+Seperating this step out allows a different id generator parameter to be passed in during testing. 
+
+
+### Routers ###
+
+The cat & breed routers are subsets of routes to handle GET, PUT, POST, DEL requests on (/cats) and (/breeds) paths, respectively. 
+The router files only contain middleware functions; all other functions are imported in. This eases monitoring of the middleware stack control flow.
+Routes invoke corresponding methods of the cat & breed class repositories which retrieve/update information directly from the database.  
+A successful route request returns object data as requested/updated/newly-created. An unsuccessful request returns a 404 NOT FOUND error. 
+Both routers contain middleware functions to verify passed id and object parameters, explained below. 
+Router instances are mounted in the main server file. 
+
+```javascript
+// GET route all 
+        this.catsRouter.get('/', (req, res, next) => {  
+            const cats = this.catRepository.getAllCats();  
+            res.json({"cats": cats}); 
+        });
+```
+
+## Cat Router ##
+
+Cat Router registers a full set of routes to enable CRUD functionality, in addition to the following middleware functions used to verify parameters:
+
+isIdValid() checks if an id is a valid shortid.
+
+checkObject() executes a pipeline of check functions to verify that the request object is populated with valid keys and values.
+An invalid id or object will generate a 400 BAD REQUEST error.
+
+## Breed Router ##
+
+Breed Router registers a full set of routes to enable CRUD functionality, in addition to methods to verify parameters:
+
+isBreedIdNum() checks if an id is a number data type. 
+
+checkBreedObj() executes a pipeline of check functions to verify that the request object is populated with valid keys and values.
+Only keys matching "name" & "description" will be accepted. Only values of 'string' data type will be accepted. 
+An invalid id or object will generate a 400 BAD REQUEST error.
+
+
+### Repository classes ###
+
+The cat & breed repository classes are pre-populated with properties containing cat & breeds objects, respectively. 
+Each class contains getter and setter methods to retrieve and update data as a complete set. 
+The subsequent methods employ the getter and setter methods, manipulating data further depending on the request.
+
```

Comment:
It's probably worth mentioning that the repository just uses an in-memory array to store cat/breed information. You can explain that in a real production application you would use a SQL database such as [sqlite](https://www.sqlite.org/index.html) to persist the data in tables.